pkg/controller/nodes/subworkflow/handler.go

```diff
@@ -94,18 +94,24 @@ func (w *workflowNodeHandler) Abort(ctx context.Context, nCtx handler.NodeExecut
 	wf := nCtx.Workflow()
 	wfNode := nCtx.Node().GetWorkflowNode()
 	if wfNode.GetSubWorkflowRef() != nil {
-		return w.subWfHandler.HandleAbort(ctx, nCtx, wf, *wfNode.GetSubWorkflowRef())
+		return w.subWfHandler.HandleAbort(ctx, nCtx, wf, *wfNode.GetSubWorkflowRef(), reason)
 	}
 
 	if wfNode.GetLaunchPlanRefID() != nil {
-		return w.lpHandler.HandleAbort(ctx, wf, nCtx.Node())
+		return w.lpHandler.HandleAbort(ctx, wf, nCtx.Node(), reason)
 	}
 	return nil
 }
 
-func (w *workflowNodeHandler) Finalize(ctx context.Context, executionContext handler.NodeExecutionContext) error {
-	logger.Debugf(ctx, "WorkflowNode::Finalizer: nothing to do")
-	return nil
+func (w *workflowNodeHandler) Finalize(ctx context.Context, nCtx handler.NodeExecutionContext) error {
+	if nCtx.Node().GetWorkflowNode().GetSubWorkflowRef() == nil {
+		logger.Infof(ctx, "finalize for workflowNode:LaunchPlanRef is a noop (no finalize required)")
+		return nil
+	}
+	logger.Info(ctx, "Finalizing Subworkflow")
+	wf := nCtx.Workflow()
```

Comment:
this is deprecated?

Reply:
yes because I want to move away from using a workflow and move to using just the contextual structure